### ISSUE TITLE
Menu: Proper role and state is not announced to the screen reader users.

### ISSUE THREAD

**Issue by MayuriNaik** [microsoft/vscode#77075]
**Environment Details:**
VSCode Version : 1.35.6 (user setup)
OS Version : Win10

GitHubTags: #A11yMAS; #A11yTCS; #Win32; #DesktopApp; #A11y_VSCodeEditor; #MAS4.1.2; #Benchmark; #A11ySev2

**Additional Details:**
MAS Violated : MAS 4.1.2
ScreenReader Versions : NVDA2019.1.1, JAWS 2019

**Repro Steps:**
1.	Open Visual studio code editor.
2.	Using "Alt+F" key navigate to the "File" menu bar option.
3.	Using "Alt+E" key navigate to the "Edit" menu bar option.
4.	Using "Alt+S" key navigate to the "Selection" menu bar option.
5.	Using "Alt+V" key navigate to the "View" menu bar option.
6.	Using "Alt+G" key navigate to the "Go" menu bar option.
7.	Using "Alt+D" key navigate to the "Help" menu bar option.
8.	Using "Alt+T" key navigate to the "Tasks" menu bar option.
9.	Using "Alt+H" key navigate to the "Help" menu bar option.
10.	Using "Down" arrow key navigate to the menu items in the file menu option.
11.	Start screen reader.
12.	Navigate to “New file” and listen if its role is announced or not.
13.	Navigate to “Open recent” and activate it, then listen if its state is announced or not.

**Actual:**
Issue 1:
When screen reader users navigate the for “New file”, “New window”, “Open File”, etc. no proper roles are been announced.

Issue 2:
When screen reader users navigate and activate the “Open recent”, “Preferences”, etc. some content gets expanded/collapsed but the state is not announced to the screen reader users.

**Expected:**
Issue 1:
The proper role as “menuitem” should be announced to the screen reader user for “New file”, “New window”, “Open File”, etc.

Issue 2:
The expand/collapse state should be announced to the screen reader users for “Open recent”, “Preferences”, etc.

**Recommendations:**
Refer below link which is repository of bug fixes code snippets:
[https://microsoft.sharepoint.com/teams/msenable/mas/pages/browse-fixes.aspx](https://microsoft.sharepoint.com/teams/msenable/mas/pages/browse-fixes.aspx)

**User Impact:**
If the role and state of an interactive element is not announced to the screen reader users the then they might not know how to interact with them.

**MAS Reference:**
MAS 1.3.1 - [https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={54f28d1f-a2d1-4dcd-84e1-5c9b87e8aba4}](https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={54f28d1f-a2d1-4dcd-84e1-5c9b87e8aba4}) 

MAS 4.1.2 - [https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={248054a6-5e68-4771-9e1e-242fb5025730}](https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={248054a6-5e68-4771-9e1e-242fb5025730}) 

**Attachment for Reference:**
[VSCode_MenubarOptions_ScreenReader Proper role and state is not announced to the screen reader users..pptx](https://github.com/microsoft/vscode/files/3376689/VSCode_MenubarOptions_ScreenReader.Proper.role.and.state.is.not.announced.to.the.screen.reader.users.pptx)

**Comment by isidorn** [comment]
I see that we are assinging the role [here](https://github.com/Microsoft/vscode/blob/f06db31fcc7c0534014937b6e10316e65969a4e4/src/vs/base/browser/ui/menu/menu.ts#L413)
However we never set the `aria-expanded` 
@sbatten Can you please look into adding that and check if the role is properly set

**Comment by MayuriNaik** [comment]
Apologies updated my mistake.

**Comment by georgebatalinski** [comment]
Let me guys know if here would be the place to update. Also, not sure if this would be an issue:
https://github.com/w3c/aria/issues/454

https://github.com/microsoft/vscode/pull/79775

**Comment by sbatten** [comment]
Issue was fixed by PR contribution long ago, closing now but should be verified for completion-sake
